pkg/cmd/server/origin/auth.go

```diff
@@ -313,9 +313,12 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 				return nil, err
 			}
 
+			// Default state builder, combining CSRF and return URL handling
```

Comment:
Why combine the error handling in here?  I can see that the error reporting uses the state, but it's not clear to me why it should also be a handler in its own right as opposed to having a separate handler that uses the state.